include/secp256k1_frost.h

```diff
@@ -0,0 +1,149 @@
+#ifndef SECP256K1_FROST_H
+#define SECP256K1_FROST_H
+
+#include "secp256k1_extrakeys.h"
+#include "../src/scalar.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h>
+
+/** This code is currently a work in progress. It's not secure nor stable.  IT
+ * IS EXTREMELY DANGEROUS AND RECKLESS TO USE THIS MODULE IN PRODUCTION!
+
+ * This module implements a modified version of Flexible Round-Optimized
+ * Schnorr Threshold Signatures (FROST) by Chelsea Komlo and Ian Goldberg
+ * (https://crysp.uwaterloo.ca/software/frost/).
+ */
+
+/** A FROST secret share. Created with `secp256k1_frost_share_gen` for a
+ * specific set of signers. Secret shares should *never* be reused across
+ * multiple signer sets.
+ *
+ * This data structure is guaranteed to be a 32-byte byte array; it is a
+ * separate type from ordinary secret keys to help prevent API users confusing
+ * shares with complete keys or using the non-FROST API in place of the FROST
+ * API, which would result in mysteriously invalid signatures being produced.
+ */
+typedef struct {
+    unsigned char data[32];
+} secp256k1_frost_share;
+
+typedef struct {
+    unsigned char data[68];
+} secp256k1_frost_secnonce;
+
+typedef struct {
+    unsigned char data[134];
+} secp256k1_frost_pubnonce;
+
+typedef struct {
+    unsigned char data[132];
+} secp256k1_frost_aggnonce;
+
+typedef struct {
+    int fin_nonce_parity;
+    unsigned char fin_nonce[32];
+    secp256k1_scalar noncecoef;
+    secp256k1_scalar challenge;
+    secp256k1_scalar s_part;
+} secp256k1_frost_session_internal;
+
+typedef struct {
+    unsigned char data[133];
+} secp256k1_frost_session;
+
+typedef struct {
+    unsigned char data[36];
+} secp256k1_frost_partial_sig;
+
+/* TODO: sanity check argument order */
+
+SECP256K1_API int secp256k1_frost_share_gen(
+    const secp256k1_context *ctx,
+    secp256k1_pubkey *pubcoeff,
+    secp256k1_frost_share *shares,
```

Comment:
We should explicitly state the requirements on the communication channel, i.e. the shares are supposed to be sent through secure (authenticated and encrypted) channels to the other signers. Also, the `vss_hashes` must be authentic, etc.